examples/coap-rd.c

```diff
@@ -657,14 +659,19 @@ join(coap_context_t *ctx, char *group_name) {
     }
   }
 
-  if (ctx->endpoint) {
-    result = setsockopt(ctx->endpoint->sock.fd,
-                        IPPROTO_IPV6, IPV6_JOIN_GROUP,
-                        (char *)&mreq, sizeof(mreq) );
-    if ( result == COAP_SOCKET_ERROR ) {
-      fprintf( stderr, "join: setsockopt: %s\n", coap_socket_strerror() );
+  endpoint = ctx->endpoint;
+  while (endpoint) {
+    if (endpoint->proto == COAP_PROTO_UDP ||
+        endpoint->proto == COAP_PROTO_DTLS) {
+      result = setsockopt(endpoint->sock.fd, IPPROTO_IPV6, IPV6_JOIN_GROUP, (char *)&mreq, sizeof(mreq));
+      if ( result == COAP_SOCKET_ERROR ) {
```

Comment:
I was working on a very similar fix yesterday and came to the conclusion that it would be sufficient if any IPV6_JOIN_GROUP was successful. Therefore, I would output the error message only after the while loop in case `!mgroup_setup`.

The other thing that I was struggling with is that access to `endpoint->next`. Maybe we need another abstract iterator function here in order to make less assumption about internal structures?

Reply:
Because of the abstractions, it may be best to go for a new function called, say, coap_join_mcast_group() - which then reduces the code in the examples.

I cant do this today as I am just going out.

Reply:
Agreed, this would be best. Maybe, I can do that based on your PR within the next hours. But there is no rush, anyway.

Reply:
Creating coap_join_mcast_group() as an alternative to join() now pushed